share/wake/lib/system/remote_cache_api.wake

```diff
@@ -645,6 +694,49 @@ export def rscApiGetFileBlob ((CacheSearchBlob blobId uri): CacheSearchBlob) (pa
 
     Pass path
 
+export def disableCacheFilePath: String = ".build/tmp/disable_remote_cache"
+
+def timeoutErrorMessage: String = "rsc: Remote cache disabled due to timeout"
+def cascadeTimeoutErrorMessage: String = "rsc: Remote cache previously disabled due to timeout"
+
+def randomUUID Unit: Result String Error =
+    def uuidJob =
+        makeExecPlan ("uuidgen", "-r", Nil) Nil
+        | setPlanLabel "generate uuid"
+        | setPlanPersistence Once # Exacally one UUID per wake invocation
+        | runJobWith localRunner
+
+    require True = isJobOk uuidJob
+    else failWithError "failed to generate UUID for Wake invocation."
+
+    uuidJob
+    | getJobStdout
+
+def disableRemoteCache Unit: Result Unit Error =
+    require Pass uuid = randomUUID Unit
+    require Pass _ = write disableCacheFilePath uuid
+    Pass Unit
+
+def readTmpFile (pathStr: String): Result String Error =
+    def imp p = prim "read"
```

Comment:
Definitely add a doc comment here saying why we can't use the standard `read` (`target` *and* `Path`).